R/dose_response.R

```diff
@@ -713,8 +713,8 @@ curve_fit_controller_fitData_dataTable_to_fitData <- function(serviceDataTable)
                                                                                                                            list(list()),
                                                                                                                            TRUE,
                                                                                                                            list(list()))]
-  serviceDataTable[!exists("userFlagStatus") || is.na(userFlagStatus), userFlagStatus := ""]
-  serviceDataTable[!exists("algorithmFlagStatus") || is.na(algorithmFlagStatus), algorithmFlagStatus := ""]
+  serviceDataTable[!exists("userFlagStatus") | is.na(userFlagStatus), userFlagStatus := ""]
+  serviceDataTable[!exists("algorithmFlagStatus") | is.na(algorithmFlagStatus), algorithmFlagStatus := ""]
```

Comment:
This was actually a bug as the || would have been comparing all user flag status and all algorithmFlagStatus in the data instead of per row.  In R 4.2.0 this became a warning and in 4.3.0 it became an error:

https://cran.r-project.org/bin/windows/base/old/4.2.0/NEWS.R-4.2.0.html
> Calling && or || with either argument of length greater than one now gives a warning (which it is intended will become an error).

In 4.3.0 this became an error: https://cran.r-project.org/bin/windows/base/old/4.3.0/NEWS.R-4.3.0.html
> Calling && or || with LHS or (if evaluated) RHS of length greater than one is now always an error, with a report of the form
    'length = 4' in coercion to 'logical(1)'

I verified I can set user flag status using curve curator and verified algorithm flagged points worked correctly.